src/BloomBrowserUI/bookEdit/OverflowChecker/OverflowChecker.ts

```diff
@@ -0,0 +1,217 @@
+                // Reposition all language tips, not just the tip for this item because sometimes the edit moves other controls.
+                (<qtipInterface>$(queryElementsThatCanOverflow)).qtip('reposition');
+            }, 100); // 100 milliseconds
+            e.stopPropagation();
+        });
+
+        // Add another handler so that when the user resizes an origami pane, we check the overflow again
+        $(container).find(".split-pane-component-inner").bind('_splitpaneparentresize', function () {
+            var $this = $(this);
+            $this.find(queryElementsThatCanOverflow).each(function () {
+                OverflowChecker.MarkOverflowInternal(this);
+            });
+        });
+
+        // Right now, test to see if any are already overflowing
+        editablePageElements.each(function () {
+            OverflowChecker.MarkOverflowInternal(this);
+        });
+    }
+
+    // Actual testable determination of Type I overflow or not
+    // 'public' for testing
+    public static IsOverflowingSelf(element: HTMLElement):boolean {
+        // Ignore Topic divs as they are chosen from a list
+        if (element.hasAttribute('data-book') && element.getAttribute('data-book') == "topic") {
+            return false;
+        }
+        // If css has "overflow: visible;", scrollHeight is always 2 greater than clientHeight.
+        // This is because of the thin grey border on a focused input box.
+        // In fact, the focused grey border causes the same problem in detecting the bottom of a marginBox
+        // so we'll apply the same 'fudge' factor to both comparisons.
+        var focusedBorderFudgeFactor = 2;
+
+        //The "basic book" template has a "Just Text" page which does some weird things to get vertically-centered
+        //text. I don't know why, but this makes the clientHeight 2 pixels larger than the scrollHeight once it
+        //is beyond its minimum height. We can detect that we're using this because it has this "firefoxHeight" data
+        //element. This problem also shows up (and is detectable the same way) in Big Book. Except it turns out the
+        //number of pixels to fudge is related to the point size. I think at base it's a preferred line spacing issue.
+        var growFromCenterVerticalFudgeFactor = 0;
+        if ($(element).data('firefoxheight')) {
+            var fontSizeRemnant = new StyleEditor("/bloom/bookEdit").GetCalculatedFontSizeInPoints(element) - 22;
+            if (fontSizeRemnant > 0) {
+                growFromCenterVerticalFudgeFactor = (fontSizeRemnant / 5) + 1;
+            }
+        }
+
+        //In the Picture Dictionary template, all words have a scrollHeight that is 3 greater than the client height.
+        //In the Headers of the Term Intro of the SHRP C1 P3 Pupil's book, scrollHeight = clientHeight + 6!!! Sigh.
+        // the focussedBorderFudgeFactor takes care of 2 pixels, this adds one more.
+        var shortBoxFudgeFactor = 4;
```

Comment:
So you're saying that it was being applied everywhere even before you pulled overflowchecker into its own file?